analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/EmptyCollectionsShouldNotBeEnumerated.cs

```diff
@@ -3,617 +3,391 @@
-    }
-
-    public void ConstructorWithEnumerable_Applies_NonEmpty()
-    {
-        var list = new ObservableCollection<int>(items);
-        list.Clear(); // Compliant
-    }
-
-    public void ConstructorWithInitializer_Applies_NonEmpty()
-    {
-        var list = new ObservableCollection<int> { 1, 2, 3 };
-        list.Clear(); // Compliant
-    }
-
-    public void ConstructorWithEmptyInitializer_Applies_Empty()
-    {
-        var list = new ObservableCollection<int> { };
-        list.Clear(); // FIXME Non-compliant
-    }
-
-    public void Other_Initialization_Applies_No_Constraints()
-    {
-        var list = Factory();
-        list.Clear(); // Compliant, we don't know anything about the list
-    }
-
-    public void Methods_Raise_Issue()
-    {
-        var list = new ObservableCollection<int>();
-        list.Clear(); // FIXME Non-compliant
-        list.IndexOf(1); // FIXME Non-compliant
-        list.Move(1, 0); // FIXME Non-compliant
-        list.Remove(1); // FIXME Non-compliant
-        var x = list[5]; // FIXME Non-compliant
-        list[5] = 5; // FIXME Non-compliant
-    }
-
-    public void Methods_Ignored()
-    {
-        var list = new ObservableCollection<int>();
-        list.GetHashCode();
-        list.Equals(items);
-        list.GetType();
-        list.ToString();
-        var a = list.Count;
-        list.CollectionChanged += List_CollectionChanged;
-    }
-
-    private void List_CollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
-    {
-        throw new NotImplementedException();
-    }
-
-    public void Methods_Set_NotEmpty()
+    public void ExtensionMethods_Should_Not_Raise()
     {
-        var list = new ObservableCollection<int>();
-        list.Add(1);
-        list.Clear(); // Compliant, will normally raise
-
-        list = new ObservableCollection<int>();
-        list.Insert(0, 1);
-        list.Clear(); // Compliant
+        var list = new List<int>();
+        list.Any();     // Compliant
     }
 
     public void ExtensionMethods_Remove_Constraints()
     {
-        var list = new ObservableCollection<int>();
-        list.Any(); // Compliant
-        list.Clear(); // Compliant, this will normally raise
-    }
-
-    public void PassingToConstructor_Removes_Constraints()
-    {
-        var list = new ObservableCollection<int>();
-        new Foo(1, 2, list); // Compliant
-        list.Clear(); // Compliant, this will normally raise
-    }
-}
-
-class ArrayTests
-{
-    public void Zero_Length_Applies_Empty()
-    {
-        var list = new int[0];
-        list.Clone(); // FIXME Non-compliant
-    }
-
-    public void NonZero_Length_Applies_NonEmpty()
-    {
-        var list = new int[5];
-        list.Clone(); // Compliant
-    }
-
-    public void Initializer_Applies_NonEmpty()
-    {
-        var list = new int[] { 1, 2, 3 };
-        list.Clone(); // Compliant
+        var list = new List<int>();
+        list.Any();
+        list.Clear();   // Noncompliant FP
```

Comment:
I'd argue that this one should be TP, because `Any` is a well-known harmless method.

I'd split this testcase into two. One with `Any` that should be `Noncompliant` and one with `list.CustomExtension()` that is a FP.

The implementation then should reset state on extension invocation.  Except well-known collection extensions like `Any` (and friends)